spring-rabbit/src/test/java/org/springframework/amqp/rabbit/retry/MissingIdRetryTests.java

```diff
@@ -223,25 +180,19 @@ public void testWithIdAndSuccess() throws Exception {
 		template.send("retry.test.exchange", "retry.test.binding", message);
 		try {
 			assertThat(cdl.await(30, TimeUnit.SECONDS)).isTrue();
-			Map map = (Map) new DirectFieldAccessor(cache).getPropertyValue("map");
-			await().until(map::isEmpty);
-			ArgumentCaptor putCaptor = ArgumentCaptor.forClass(Object.class);
-			ArgumentCaptor getCaptor = ArgumentCaptor.forClass(Object.class);
-			ArgumentCaptor removeCaptor = ArgumentCaptor.forClass(Object.class);
-			verify(cache, times(2)).put(putCaptor.capture(), any(RetryContext.class));
-			verify(cache, times(2)).get(getCaptor.capture());
-			verify(cache, atLeast(2)).remove(removeCaptor.capture());
-			verify(cache, atMost(4)).remove(removeCaptor.capture());
-			logger.debug("puts:" + putCaptor.getAllValues());
-			logger.debug("gets:" + putCaptor.getAllValues());
-			logger.debug("removes:" + removeCaptor.getAllValues());
+			await().atMost(Duration.ofSeconds(3)).until(cache::isEmpty);
```

Comment:
Might not be enough on CI/CD.
I'm totally OK with those `30 seconds` by default.